pkg/apis/ceph.rook.io/v1/network.go

```diff
@@ -56,7 +69,7 @@ func ValidateNetworkSpec(clusterNamespace string, spec NetworkSpec) error {
 	}
 
 	if !spec.AddressRanges.IsEmpty() {
-		if !spec.IsMultus() && !spec.IsHost() {
+		if !spec.IsMultus() && spec.Provider != NetworkProviderHost {
```

Comment:
I don't understand why this change is needed here. 

I think that this condition no longer works correctly if the user has `HostNetwork: true`, because the legacy behavior is no longer also supported.

Reply:
also note that the signature of `IsHost()`has changed to include an additional Error return